Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```
@@ -2497,3 +2499,16 @@ bool CLuaPedDefs::killPedTask(CClientPed* ped, taskType taskType, std::uint8_t t
             return false; 
     }
 }
+
+bool CLuaPedDefs::PedSay(CClientPed* ped, int speechId, std::optional<float> probabilty)
+{
+    auto speechContextId = static_cast<ePedSpeechContext>(speechId);
+    if (speechContextId < ePedSpeechContext::NOTHING || speechContextId >= ePedSpeechContext::NUM_PED_CONTEXT)
+        return false;
+
+    if (probabilty.has_value() && probabilty < 0.0f)
+        return false;
+
+    ped->Say(speechContextId, probabilty.value_or(1.0f));
+    return true;
+}
```

You should throw an exception if the user/script provides invalid arguments.